sim.c

```diff
@@ -615,8 +615,28 @@ static void enqueue_movement(struct board *board, struct movement m)
     board->movements.movements[board->movements.length++] = m;
 }
 
-static void move_atom(struct board *board, atom *a, struct vector position)
+static bool movements_equal(struct movement *a, struct movement *b)
 {
```

Comment:
The movement struct is kind of confusing because it includes redundant data for the reverse-engineered collision detection code.  The actual grid movement only uses `rotation` and `translation`, so I believe it's enough to do
```c
return m1.rotation == m2.rotation && vectors_equal(m1.translation, m2.translation);
```